shared/js/project.js

```diff
@@ -82,20 +87,68 @@ var projects = module.exports = {
     return isEditing;
   },
 
-  init: function () {
+  // Whether the current level is a project level (i.e. at the /projects url).
+  isProjectLevel: function() {
+    return (appOptions.level && appOptions.level.isProjectLevel);
+  },
+
+  isEmbedded: function() {
+    return (
+        (appOptions.level && appOptions.level.projectTemplateLevelName) ||
+        appOptions.app === 'applab' ||
+        appOptions.isExternalProjectLevel
+    );
+  },
+
+  shouldUpdateHeaders: function() {
+    return !appOptions.isExternalProjectLevel;
+  },
+
+  showProjectHeader: function() {
+    if (this.shouldUpdateHeaders()) {
+      dashboard.header.showProjectHeader();
+    }
+  },
+
+  showMinimalProjectHeader: function() {
+    if (this.shouldUpdateHeaders()) {
+      dashboard.header.showMinimalProjectHeader();
+    }
+  },
+
+  showProjectLevelHeader: function() {
+    if (this.shouldUpdateHeaders()) {
+      dashboard.header.showProjectLevelHeader();
+    }
+  },
+
+  //////////////////////////////////////////////////////////////////////
+  // End of properties and callbacks.
+  //////////////////////////////////////////////////////////////////////
+
+  /**
+   *
+   * @param {Object} handler Object containing callbacks provided by caller.
+   * @param {Function} handler.setInitialLevelHtml
+   * @param {Function} handler.getLevelHtml
+   * @param {Function} handler.setInitialLevelSource
+   * @param {Function} handler.getLevelSource
+   */
+  init: function (handler) {
```

Comment:
I wonder if we can come up with a less generic name than handler. Maybe something like sourceHandler? Or a better name than source that encompasses source and html